WebContent/bank/transfer.jsp

```diff
@@ -0,0 +1,112 @@
+<%@ page language="java" contentType="text/html; charset=ISO-8859-1"
+    pageEncoding="ISO-8859-1"%>
+
+<%
+/**
+ This application is for demonstration use only. It contains known application security
+vulnerabilities that were created expressly for demonstrating the functionality of
+application security testing tools. These vulnerabilities may present risks to the
+technical environment in which the application is installed. You must delete and
+uninstall this demonstration application upon completion of the demonstration for
+which it is intended. 
+
+IBM DISCLAIMS ALL LIABILITY OF ANY KIND RESULTING FROM YOUR USE OF THE APPLICATION
+OR YOUR FAILURE TO DELETE THE APPLICATION FROM YOUR ENVIRONMENT UPON COMPLETION OF
+A DEMONSTRATION. IT IS YOUR RESPONSIBILITY TO DETERMINE IF THE PROGRAM IS APPROPRIATE
+OR SAFE FOR YOUR TECHNICAL ENVIRONMENT. NEVER INSTALL THE APPLICATION IN A PRODUCTION
+ENVIRONMENT. YOU ACKNOWLEDGE AND ACCEPT ALL RISKS ASSOCIATED WITH THE USE OF THE APPLICATION.
+
+IBM AltoroJ
+(c) Copyright IBM Corp. 2008, 2013 All Rights Reserved.
+*/
+%>  
+    
+<jsp:include page="/header.jspf"/>
+
+<div id="wrapper" style="width: 99%;">
+	<jsp:include page="membertoc.jspf"/>
+    <td valign="top" colspan="3" class="bb">
+    	<%@page import="com.ibm.security.appscan.altoromutual.model.Account"%>
+		
+		<%
+					com.ibm.security.appscan.altoromutual.model.User user = (com.ibm.security.appscan.altoromutual.model.User)request.getSession().getAttribute("user");
+				%>
+				
+		<script type="text/javascript">
+		
+			function confirminput(myform) {
+				var dbt=document.getElementById("fromAccount").value;
+				var cdt=document.getElementById("toAccount").value;
+				var amt=document.getElementById("transferAmount").value;
+				
+				if (dbt == cdt) {
+					alert("From Account and To Account fields cannot be the same.");
+					return false;
+				}
+				else if (!(amt > 0)){
+					alert("Transfer Amount must be a number greater than 0.");
+					return false;
+				}
+		
+				return true;
+			}
+
+		</script>
+		
+		<div class="fl" style="width: 99%;">
+		
+		<form id="tForm" name="tForm" method="post" action="doTransfer" onsubmit="return (confirminput(tForm));">
+		
+		<h1>Transfer Funds</h1>
+		
+		<table cellSpacing="0" cellPadding="1" width="100%" border="0">
+		  <tr>
+		    <td><strong>From Account:</strong>
+		    </td>
+		    <td>
+			  <select size="1" id="fromAccount" name="fromAccount">
+				<% 
+				for (Account account: user.getAccounts()){
+					out.println("<option value=\""+account.getAccountId()+"\" >" + account.getAccountId() + " " + account.getAccountName() + "</option>");
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg" width="20"/></sub>&nbsp; Cross-site Scripting (XSS)

Unsanitized input from `a database` flows into `println`, where it is used to render an HTML page returned to the user. This may result in a Cross-Site Scripting attack (XSS).

<h6>
Line 70 | <a href="https://cwe.mitre.org/data/definitions/79.html">CWE-79</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 592</a> | <a href="https://learn.snyk.io/lesson/xss/">Learn more about this vulnerability</h6>

<details>

<summary><b>Data flow: 19 steps</b></summary>

`Step 1 - 3` https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L281

`Step 4 - 5` [src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L283](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L283)

`Step 6` [src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L284](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L284)

`Step 7 - 9` [src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L287](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/src/com/ibm/security/appscan/altoromutual/util/DBUtil.java#L287)

`Step 10` [src/com/ibm/security/appscan/altoromutual/model/User.java#L78](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/src/com/ibm/security/appscan/altoromutual/model/User.java#L78)

`Step 11 - 12` [WebContent/bank/transfer.jsp#L69](https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/bank/transfer.jsp#L69)

`Step 13 - 19` https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/bank/transfer.jsp#L70

</details>